test/tests/functional/pbs_pbsnodes.py

```diff
@@ -107,3 +171,70 @@ def test_pbsnodes_av(self):
         self.logger.info("Before: " + prev + ". After: " + now + ".")
         self.assertEquals(prev.strip(), now.strip(),
                           'Last used time mismatch after server restart')
+
+    @skipOnCpuSet
+    @skipOnCray
+    def test_pbsnodes_as_user(self):
+        """
+        Validate default values of node attributes for non-root user
+        """
+        self.common_setUp()
+        self.server.manager(MGR_CMD_DELETE, NODE, id="",
+                            sudo=True, expect=True)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom.shortname)
+        self.server.expect(NODE, {'state': 'free'})
+
+        attr_dict = {}
+        expected_attrs = self.get_newnode_attrs(TEST_USER)
+        command = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                               'bin', 'pbsnodes -a')
+        ret = self.du.run_cmd(self.server.hostname, command)
+        attr_list = ret['out']
+        list_len = len(attr_list) - 1
+        for i in range(1, list_len):
+            attr = attr_list[i].split('=')[0].strip()
+            val = attr_list[i].split('=')[1].strip()
+            attr_dict[attr] = val
+
+        # comparing the pbsnodes -a output with expected result
+        for attr in expected_attrs:
+            self.assertEqual(expected_attrs[attr], attr_dict[attr])
+
+        self.verify_node_dynamic_val(attr_dict['last_state_change_time'],
+                                     attr_dict['resources_available.ncpus'],
+                                     attr_dict['pcpus'], attr_dict['sharing'],
+                                     attr_dict['resources_available.mem'])
+
+    @tags('smoke')
+    @skipOnCpuSet
+    @skipOnCray
+    def test_pbsnodes_as_root(self):
+        """
+        Validate default values of node attributes for root user
+        """
+        self.common_setUp()
+        self.server.manager(MGR_CMD_DELETE, NODE, id="",
+                            sudo=True, expect=True)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom.shortname)
+        self.server.expect(NODE, {'state': 'free'})
```

Comment:
just a thought, since your tests needs this special setUp may be above 3 lines can get defined under common_setUp.